test/test.js

```diff
@@ -0,0 +1,84 @@
+var nock   = require('nock');
+
+var logic = require('../logic.js');
+
+function responseLog(response) {
+  console.log('   RESPONSE SENT: ' +  response);
+}
+
+var ghUser = nock('https://api.github.com')
+              .get('/user')
+              .reply(200, {
+                login: 'testuser',
+                id: 1234567,
+                avatar_url: '',
+                gravatar_id: '',
+                url: 'https://api.github.com/users/testuser',
+                html_url: 'https://github.com/testuser',
+                followers_url: 'https://api.github.com/users/testuser/followers',
+                following_url: 'https://api.github.com/users/testuser/following{/other_user}',
+                gists_url: 'https://api.github.com/users/testuser/gists{/gist_id}',
+                starred_url: 'https://api.github.com/users/testuser/starred{/owner}{/repo}',
+                subscriptions_url: 'https://api.github.com/users/testuser/subscriptions',
+                organizations_url: 'https://api.github.com/users/testuser/orgs',
+                repos_url: 'https://api.github.com/users/testuser/repos',
+                events_url: 'https://api.github.com/users/testuser/events{/privacy}',
+                received_events_url: 'https://api.github.com/users/testuser/received_events',
+                type: 'User',
+                site_admin: false,
+                name: 'Test User',
+                company: null,
+                blog: null,
+                location: null,
+                email: null,
+                hireable: false,
+                bio: null,
+                public_repos: 0,
+                public_gists: 0,
+                followers: 0,
+                following: 0,
+                created_at: '2016-00-00T00:13:37Z',
+                updated_at: '2016-03-14T00:15:09Z'
+              });
+
+describe('welcome()', function() {
+  it('Should be able to authenticate with GitHub', function(done) {
+    logic.welcome('token', function(err, user) {
+      done(err);
```

Comment:
so you can use assert and test if the function behaves as expected.

`assert(user=='testuser')`

there are some libraries that you can use as well. to write tests more elegantly than using basic 'assert'. see http://chaijs.com/